R/identify_uniform_segment.R

```diff
@@ -20,16 +20,88 @@
 #'
 #' @export
 identify_uniform_segment <- function(
+    x,
+    metric = c("jaccard", "intersection", "ks", "mse", "chisq"),
+    threshold = 0.5,
+    stepsize = 1,
+    max_sd_size = 1
+){
+  UseMethod("identify_uniform_segment")
+}
+
+#' @exportS3Method identify_uniform_segment numeric
+identify_uniform_segment.numeric <- function(
+    x,
+    metric = c("jaccard", "intersection", "ks", "mse", "chisq"),
+    threshold = 0.5,
+    stepsize = 1,
+    max_sd_size = 1
+){
+  interval_midpoint <- seq(1, length(x), 1)
+  identify_uniform_segment_helper(
+    x = x,
+    interval_start = interval_midpoint - 0.5,
+    interval_end = interval_midpoint + 0.5,
+    interval_midpoint = interval_midpoint,
+    metric = metric,
+    threshold = threshold,
+    stepsize = stepsize,
+    max_sd_size = max_sd_size
+  )
+}
+
+#' @exportS3Method identify_uniform_segment Histogram
+identify_uniform_segment.Histogram <- function(
+    x,
+    metric = c("jaccard", "intersection", "ks", "mse", "chisq"),
+    threshold = 0.5,
+    stepsize = 1,
+    max_sd_size = 1
+){
+  identify_uniform_segment_helper(
+    x = x$histogram_data,
+    interval_start = x$interval_start,
+    interval_end = x$interval_end,
+    interval_midpoint = find_midpoint(x),
+    metric = metric,
+    threshold = threshold,
+    stepsize = stepsize,
+    max_sd_size = max_sd_size
+  )
+}
+
+#' @exportS3Method identify_uniform_segment GenomicHistogram
```

Comment:
Use `@rdname identify_uniform_segment` to link all pages together. See other comment with more info.